### ISSUE TITLE
sh.vim: Feature request - Better processing of bash's substring expansion

### ISSUE THREAD

**Issue by norman-abramovitz** [vim/vim#1437]
A valid substring expansion that can be differentiated from the :- expansion should not show as an error. 
```bash
#!/bin/bash
f() {
  local last=${1: -1:1}
  echo ${last}
}
f abc
```

**Comment by cecamp** [comment]
Please try v168 of syntax/sh.vim -- I don't see a problem with your example with it.
Regards,
Chip Campbell

**Comment by norman-abramovitz** [comment]
Tried out 168 and still seeing the error. 

Uploading some screenshots showing the error colorization.  Also uploaded some highlight colorization showing that colorization is an error colorization.

<img width="277" alt="screen shot 2017-02-06 at 10 20 42 am" src="https://cloud.githubusercontent.com/assets/13754664/22660378/25575814-ec56-11e6-9c10-8a9bd17c47f2.png">

<img width="353" alt="screen shot 2017-02-06 at 9 51 29 am" src="https://cloud.githubusercontent.com/assets/13754664/22659965/8044a27e-ec54-11e6-8b28-c75cc4ec3f89.png">
<img width="376" alt="screen shot 2017-02-06 at 9 49 16 am" src="https://cloud.githubusercontent.com/assets/13754664/22659977/8bf8175e-ec54-11e6-81e3-76654ad230c4.png">

**Comment by lacygoill** [comment]
Fixed by e4a3bcf28d92d0bde9ca227ccb40d401038185e5.
